File: mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/pinpointclient/PinPointClientMvpView.java
package com.mifos.mifosxdroid.activity.pinpointclient;

import androidx.annotation.Nullable;

import com.mifos.mifosxdroid.base.MvpView;
import com.mifos.objects.client.ClientAddressResponse;

import java.util.List;

/**
 * Created by Rajan Maurya on 08/06/16.
 */
public interface PinPointClientMvpView extends MvpView {

    void showUserInterface();

    void requestPermission(int requestCode);

    void showClientPinpointLocations(List<ClientAddressResponse> clientAddressResponses);

    void showFailedToFetchAddress();

    void showEmptyAddress();

    void showPlacePiker(int requestCode);

    void showProgressDialog(boolean show, @Nullable Integer message);

    void showMessage(int message);

    void updateClientAddress(int message);
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/pinpointclient/PinPointClientMvpView.kt
package com.mifos.mifosxdroid.activity.pinpointclient

import com.mifos.mifosxdroid.base.MvpView
import com.mifos.objects.client.ClientAddressResponse

/**
 * Created by Rajan Maurya on 08/06/16.
 */
interface PinPointClientMvpView : MvpView {
    fun showUserInterface()
    fun requestPermission(requestCode: Int)
    fun showClientPinpointLocations(clientAddressResponses: List<ClientAddressResponse>)
    fun showFailedToFetchAddress()
    fun showEmptyAddress()
    fun showPlacePiker(requestCode: Int)
    fun showProgressDialog(show: Boolean, message: Int?)
    fun showMessage(message: Int)
    fun updateClientAddress(message: Int)
}